packages/agent/src/app-data-store.ts

```diff
@@ -0,0 +1,374 @@
+   */
+  initialize(options: { passphrase: string, keyPair: Web5Crypto.CryptoKeyPair }): Promise<void>;
+
+  /**
+   * Creates an encrypted backup of the current state of `AppData` and
+   * returns a Promise that resolves to an `AppDataBackup` object.
+   */
+  backup(options: { passphrase: string }): Promise<AppDataBackup>;
+
+  /**
+   * Restores `AppData` to the state in the provided `AppDataBackup` object.
+   * It requires a passphrase to decrypt the backup and returns a Promise that
+   * resolves to a boolean indicating whether the restore was successful.
+   */
+  restore(options: { backup: AppDataBackup, passphrase: string }): Promise<boolean>;
+
+  /**
+   * Locks the `AppDataStore`, secured by a passphrase
+   * that must be entered to unlock.
+   */
+  lock(): Promise<void>;
+
+  /**
+   * Attempts to unlock the `AppDataStore` with the provided
+   * passphrase.  It returns a Promise that resolves to a
+   * boolean indicating whether the unlock was successful.
+   */
+  unlock(options: { passphrase: string }): Promise<boolean>;
+
+  /**
+   * Attempts to change the passphrase of the `AppDataStore`.
+   * It requires the old passphrase for verification and returns
+   * a Promise that resolves to a boolean indicating whether the
+   * passphrase change was successful.
+   */
+  changePassphrase(options: { oldPassphrase: string, newPassphrase: string }): Promise<boolean>;
+}
+
+export type AppDataVaultOptions = {
+  keyDerivationWorkFactor?: number;
+  store?: KeyValueStore<string, any>;
+}
+
+export class AppDataVault implements AppDataStore {
+  private _keyDerivationWorkFactor: number;
+  private _store: KeyValueStore<string, any>;
+  private _vaultUnlockKey = new Uint8Array();
+
+  constructor(options?: AppDataVaultOptions) {
+    this._keyDerivationWorkFactor = options?.keyDerivationWorkFactor ?? 650_000;
+    this._store = options?.store ?? new MemoryStore();
+  }
+
+  async backup(options: { passphrase: string }): Promise<AppDataBackup> {
+    console.log(options);
+    return null as any;
+  }
+
+  async changePassphrase(options: { oldPassphrase: string, newPassphrase: string }): Promise<boolean> {
+    console.log(options);
+    return null as any;
+  }
+
+  private async generateVaultUnlockKey(options: {
+    passphrase: string,
+    salt: Uint8Array
+  }): Promise<Uint8Array> {
+    const { passphrase, salt } = options;
+
+    /** The salt value derived in Step 3 and the passphrase entered by the
+     * end-user are inputs to the PBKDF2 algorithm to derive a 32-byte secret
+     * key that will be referred to as the Vault Unlock Key (VUK). */
+    const vaultUnlockKey = await pbkdf2Async(
+      sha512,                // hash function
+      passphrase,            // password
+      salt,                  // salt
+      {
+        c     : this._keyDerivationWorkFactor, // key derivation work factor
+        dkLen : 32           // derived key length, in bytes
+      }
+    );
```

Comment:
This `pbkdf2Async` call is deadlocking on React Native. 

I remember @shamilovtim and @thehenrytsai previously ran into some issues with `@noble` packages on React Native. Do either of you remember what exactly the issue was back then? If I recall correctly, it was something to do with their bundling process, but don't remember the exact details. I just remember that an update to the packages was needed in order to address.

I see that `@web5/crypto` has pre-2.x packages in it, perhaps that's contributing?
```
  "dependencies": {
    "@noble/ciphers": "0.1.4",
    "@noble/curves": "1.1.0",
    "@noble/hashes": "1.3.1",
    ...
  },
```

Reply:
Does this still need a look? Can try to look at this today

Reply:
This is still an issue, and I haven't been able to figure it out yet. I noticed in the README file for this package there's a required polyfill in order to use it: https://github.com/paulmillr/noble-hashes#usage

I added that polyfill, and I'm still getting a deadlock in the app. Would appreciate any help here!

Reply:
Pretty sure I know why this is.  The default is to execute a sufficient number of iterations to derive a key from the password based input that it is very unlikely for an attacker to be able to brute-force.  This takes a lot of compute time so that may be the issue.  It can take 5-10 seconds so perhaps it is exceeding some built-in timer for React Native?

Can you try again with `keyDerivationWorkFactor` set to `1`?

Reply:
That worked! We can go ahead and resolve this one. 

If anyone would like to follow along with integrating this into the React Native identity agent, I've pushed a branch over there with all of my work so far! https://github.com/TBD54566975/web5-wallet/tree/amika/identity-agent